lib/bolt/pal/yaml_plan.rb

```diff
@@ -117,23 +110,23 @@ def validate
           validate_step_keys(step, step_number)
 
           begin
-            step.each { |k, v| validate_puppet_code(k, v) }
+            step.body.each { |k, v| validate_puppet_code(k, v) }
```

Comment:
This seems like this validation should move to the step now that it's it's own object.